frontend/src/hooks/useToast.ts

```diff
@@ -0,0 +1,38 @@
+import { PropsWithChildren } from 'react'
+import {
+  RenderProps,
+  useToast as useChakraToast,
+  UseToastOptions,
+} from '@chakra-ui/react'
+
+import { Toast, ToastProps } from '~/components/Toast/Toast'
+
+type ToastBehaviourProps = Pick<
+  UseToastOptions,
+  'duration' | 'position' | 'render'
+>
+
+// onClose is provided by the chakra hook and should not be exposed to clients
+export type UseToastProps = Omit<ToastBehaviourProps & ToastProps, 'onClose'>
+
+type UseToast = () => (props: UseToastProps) => void
+
+export const useToast: UseToast = () => {
```

Comment:
should we also allow useToast to accept initial props like in chakra's own useToast?

```ts
export const useToast: UseToast = (initialProps) => {
  const toast = useChakraToast(initialProps)
  ...
}
```

Reply:
good catch! edited